File: docs/site/samples/typescript/examples/guides_loader_miss.ts
/**
 * Loader guide — a missing key rejects with PromptLoadError (load_not_found),
 * distinct from LoadError (the parse/shape error thrown on malformed YAML).
 *
 * `catch (e) { if (e instanceof PromptLoadError) }` does NOT catch a malformed-YAML `LoadError`.
 */

import assert from "node:assert/strict";
import { test } from "node:test";
import {
	LOAD_NOT_FOUND,
	LoadError,
	MemoryLoader,
	Prompt,
	PromptLoadError,
} from "prompting-press";

test("missing key rejects with PromptLoadError (load_not_found)", async () => {
	const loader = new MemoryLoader({});

	// A missing key rejects with PromptLoadError — not a parse error.
	await assert.rejects(
		() => loader.load("missing"),
		(err: unknown) => {
			assert.ok(err instanceof PromptLoadError);
			assert.equal(err.errors[0].code, LOAD_NOT_FOUND);
			return true;
		},
	);

	// PromptLoadError is distinct from LoadError.
	// Parsing bad YAML throws LoadError — a different type on a different path.
	assert.throws(
		() => Prompt.fromYaml("not: valid: yaml: ["),
		(err: unknown) => {
			assert.ok(err instanceof LoadError);
			return true;
		},
	);
});

File: docs/site/samples/python/examples/guides_loader_miss.py
# Copyright (C) 2024-2026 Sjors Robroek
# SPDX-License-Identifier: AGPL-3.0-only

"""Loader guide — a missing key raises PromptLoadError (load_not_found),
distinct from LoadError (the parse/shape error raised on malformed YAML).

``except PromptLoadError`` does NOT catch a malformed-YAML ``LoadError``.
"""

from prompting_press import LoadError, Prompt, PromptLoadError
from prompting_press.loader import LOAD_NOT_FOUND, MemoryLoader


def main() -> None:
    loader = MemoryLoader({})

    # A missing key raises PromptLoadError — not a parse error.
    try:
        loader.load("missing")
        assert False, "should have raised"
    except PromptLoadError as exc:
        assert exc.errors[0].code == LOAD_NOT_FOUND

    # PromptLoadError is distinct from LoadError.
    # Parsing bad YAML raises LoadError — a different type on a different path.
    try:
        Prompt.from_yaml("not: valid: yaml: [")
        assert False, "should have raised"
    except LoadError:
        pass


if __name__ == "__main__":
    main()
